uint32 area
uint32 perimeter
uint32[] center
uint32[] size
uint8[] data